### PR TITLE
Fix print effect on forever page

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Update implementation.

## Justify why they are needed

With async rendering we can't be sure about the order anymore so we got unpredictable behavior with "setInterval".

I've tested this approach and seems to work fine.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #459** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/459" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/459?utm_source=stack-comment).

File: apps/onboarding/src/components/forever/usePrintCodeEffect.ts [hunk 1]
```diff
@@ -8,27 +8,21 @@ type PrintCodeEffectParams = {
...
-    }, LOAD_CHAR_INTERVAL)
-
-    return () => {
-      window.clearInterval(handle)
+    if (charIndex >= initialCode.length) return
+
+    const tick = () => {
+      setCode(initialCode.slice(0, charIndex + 1))
+      setCharIndex(charIndex + 1)
+      console.log('ajd')
```
Comment (alebedev) [inline apps/onboarding/src/components/forever/usePrintCodeEffect.ts:19]: debug log?

File: apps/onboarding/src/components/forever/usePrintCodeEffect.ts [hunk 2]
```diff
@@ -8,27 +8,21 @@ type PrintCodeEffectParams = {
 
 export const usePrintCodeEffect = ({ initialCode }: PrintCodeEffectParams) => {
   const [code, setCode] = useState('')
```
Comment (alebedev) [inline apps/onboarding/src/components/forever/usePrintCodeEffect.ts:10]: Do we need state for code?  We can calculate it as `initialCode.slice(0, charIndex + 1)` at every render

Less state we have, less opportunity for conflicts

**Review by alebedev — APPROVED** [review]

**Review by guilhermespopolin — APPROVED** [review]
Nice. Just remove the log as Alexander mentioned and we'd be good to go.

File: apps/onboarding/src/components/forever/usePrintCodeEffect.ts (hunk 2 above)
Comment (robinandeer) [reply apps/onboarding/src/components/forever/usePrintCodeEffect.ts:10]: @alebedev hmm but what triggers those renders? we need to update state to trigger the next render, right?

**Pull request merged**